pkg/application-grid-controller/controller/deployment/node.go

```diff
@@ -32,31 +32,36 @@ func (dgc *DeploymentGridController) addNode(obj interface{}) {
 	if node.DeletionTimestamp != nil {
 		// On a restart of the controller manager, it's possible for an object to
 		// show up in a state that is already pending deletion.
-		dgc.deleteDeployment(node)
+		dgc.deleteNode(node)
 		return
 	}
 	dgs := dgc.getGridForNode(node)
 	if len(dgs) == 0 {
 		return
 	}
-	klog.V(4).Infof("Node %s added.", node.Name)
 	for _, dg := range dgs {
+		klog.V(4).Infof("Node %s(its relevant DeploymentGrid %s) added.", node.Name, dg.Name)
 		dgc.enqueueDeploymentGrid(dg)
 	}
 }
 
 func (dgc *DeploymentGridController) updateNode(oldObj, newObj interface{}) {
 	oldNode := oldObj.(*corev1.Node)
 	curNode := newObj.(*corev1.Node)
+	if curNode.ResourceVersion == oldNode.ResourceVersion {
+		// Periodic resync will send update events for all known Nodes.
+		// Two different versions of the same Node will always have different RVs.
+		return
+	}
 	labelChanged := !reflect.DeepEqual(curNode.Labels, oldNode.Labels)
 	// Only handles nodes whose label has changed.
 	if labelChanged {
-		dgs := dgc.getGridForNode(curNode)
+		dgs := dgc.getGridForNode(oldNode, curNode)
```

Comment:
GetGridForNode -> GetNodesSelector  use GridSelectorUniqKeyName as key, seems can't get wanted deployment grid?

Reply:
That's right, we need fix it.